File: app/src/main/java/com/zfdang/dimensioncam/data/PhotoDao.java
package com.zfdang.dimensioncam.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface PhotoDao {
    @androidx.room.Transaction
    @Query("SELECT * FROM photos ORDER BY createdAt DESC")
    LiveData<List<PhotoWithAnnotations>> getPhotosWithAnnotations();

    @Query("SELECT * FROM photos ORDER BY createdAt DESC")
    LiveData<List<Photo>> getAllPhotos();

    @Insert
    long insert(Photo photo);

    @Delete
    void delete(Photo photo);
    
    @Query("SELECT * FROM photos WHERE id = :id")
    Photo getPhotoById(long id);
}

File: app/src/main/java/com/dimensioncam/data/dao/PhotoDao.kt
package com.dimensioncam.data.dao

import androidx.room.*
import com.dimensioncam.data.model.Photo
import kotlinx.coroutines.flow.Flow

/**
 * Data Access Object for Photo entity
 */
@Dao
interface PhotoDao {
    
    @Query("SELECT * FROM photos ORDER BY modifiedAt DESC")
    fun getAllPhotos(): Flow<List<Photo>>
    
    @Query("SELECT * FROM photos WHERE id = :photoId")
    suspend fun getPhotoById(photoId: Long): Photo?
    
    @Query("SELECT * FROM photos WHERE id = :photoId")
    fun observePhotoById(photoId: Long): Flow<Photo?>
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertPhoto(photo: Photo): Long
    
    @Update
    suspend fun updatePhoto(photo: Photo)
    
    @Delete
    suspend fun deletePhoto(photo: Photo)
    
    @Query("DELETE FROM photos WHERE id = :photoId")
    suspend fun deletePhotoById(photoId: Long)
    
    @Query("UPDATE photos SET modifiedAt = :timestamp WHERE id = :photoId")
    suspend fun updateModifiedTime(photoId: Long, timestamp: Long = System.currentTimeMillis())
}
